yarn-project/archiver/src/archiver/kv_archiver_store/log_store.ts

```diff
@@ -391,19 +417,20 @@ export class LogStore {
   }
 
   #accumulateLogs(
-    results: (ExtendedUnencryptedL2Log | ExtendedPublicLog)[],
+    results: (ExtendedContractClassLog | ExtendedPublicLog)[],
     blockNumber: number,
     txIndex: number,
-    txLogs: (UnencryptedL2Log | PublicLog)[],
+    txLogs: (ContractClassLog | PublicLog)[],
     filter: LogFilter,
   ): boolean {
     let maxLogsHit = false;
     let logIndex = typeof filter.afterLog?.logIndex === 'number' ? filter.afterLog.logIndex + 1 : 0;
     for (; logIndex < txLogs.length; logIndex++) {
       const log = txLogs[logIndex];
-      if (!filter.contractAddress || log.contractAddress.equals(filter.contractAddress)) {
-        if (log instanceof UnencryptedL2Log) {
-          results.push(new ExtendedUnencryptedL2Log(new LogId(blockNumber, txIndex, logIndex), log));
+      // TODO(MW): fix hack for contract class
+      if (!filter.contractAddress || (log instanceof PublicLog && log.contractAddress.equals(filter.contractAddress))) {
+        if (log instanceof ContractClassLog) {
+          results.push(new ExtendedContractClassLog(new LogId(blockNumber, txIndex, logIndex), log));
```

Comment:
The contract class log here will be a siloed field array. Initially I thought there was no need to have a `.contractAddress` property on `ContractClassLog`s because by this point they would have been constrained to be from the registerer (regardless of whether they contain a valid event).
However this totally bricked a lot of tests. I added a quick hack that accepts any cc log to get them running. But maybe we should carry a contract address around with the log?

Reply:
> maybe we should carry a contract address around with the log?

This sounds sensible!